src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/SpeculationAnalyzer.cs

```diff
@@ -363,6 +363,14 @@ protected override bool ReplacementChangesSemanticsForNodeLanguageSpecific(Synta
                     var originalExpressionType = originalExpressionTypeInfo.Type;
                     var newExpressionType = newExpressionTypeInfo.Type;
 
+                    // A conditional expression may have no type of it's own, but can be converted to a type if there's target-typed
+                    // conditional expressions in play; in this case we'll use the converted type.
+                    if (newExpressionType == null &&
+                        this.SpeculativeSemanticModel.GetConversion(newExpression, this.CancellationToken).IsConditionalExpression)
+                    {
+                        newExpressionType = newExpressionTypeInfo.ConvertedType;
+                    }
```

Comment:
so, unrelated to this change, i'm really for extracting out this entire section related to conditional exprs into its own method.  this is a lot of code in an enormous mehtod.